### PR TITLE
102: submission attachment upload with submission.create

### LINKED ISSUE #102: Extend submission.create to support attachments
I sometimes have to pull data from a device and send it to a server. I'd like to script that with pyODK but there's no easy way to create a submission with attachments. It'd be nice to have that.

Here's what I'm doing...
```python
#!/usr/bin/env python3

from pathlib import Path
import requests

SUBMISSION_URL = "https://foo.getodk.cloud/v1/key/1234567890/projects/1/submission"
INSTANCES_PATH = Path("/Users/me/Desktop/instances")

def submit_instances(folder_path):
    files = {}

    for file in folder_path.iterdir():
        if file.is_file():
            file_content = file.read_bytes()
            
            if file.suffix == ".xml":
                print(file.name)
                files["xml_submission_file"] = ("instance.xml", file_content)
            else:
                files[file.name] = (file.name, file_content)

    response = requests.post(
        SUBMISSION_URL,
        files=files,
        headers={"X-OpenRosa-Version": "1.0"}
    )

    if response.status_code != 201:
        print(response.text)

for folder in INSTANCES_PATH.iterdir():
    if folder.is_dir():
        submit_instances(folder)
```

### PR DESCRIPTION
**Pull request by lindsay-stevens**
Closes #102

- accepts a list of paths and sends in chunks to account for large files
  - adds setting on session object for blocksize, although it is the same as default it allows customisation for advanced users
- fix: entity_list test fixture creation flow, noticed when running tests against a new Central instance (dev)

#### What has been done to verify that this works as intended?

Added unit and integration tests, plus some manual testing to check behaviour mentioned below.

#### Why is this the best possible solution? Were any other approaches considered?

In reference to #102 the code could now look like:

```python
#!/usr/bin/env python3

from pathlib import Path
from pyodk.client import Client

client = Client()
for folder in Path("/Users/me/Desktop/instances").iterdir():
    if folder.is_dir():
        client.submissions.create(
            project_id=1,
            form_id="my_form",
            xml=(folder / "instance.xml").read_text(),
            attachments=(f for f in folder.iterdir() if f.is_file() and f.name != "instance.xml")
        )
```

Also considered:

- send the files without chunking
  - pros: easiest
  - cons:
    - possible memory issues for sending large files
    - possible bug in Central where "Content-Type" not read for chunked transfers which means all files default to "application/octet-stream" even though pyodk is sending a mime type (guessed from file name). The content-type is saved as expected when the file isn't chunked. Not sure of any impact on Collect/Enketo/Webforms if the content-type is wrong e.g. are images not displayed? Asked about this in Slack.
- allowing more customisation e.g. by exposing upload params via a SubmissionAttachmentSettings object.
  - pros: allows fine tuning upload of each file: content-type, encoding, chunking
  - cons:
    - may cause awkward backwards compatibility scenarios in future if options or defaults need to change etc
    - may not cover all advanced use cases anyway, e.g. parallel uploads, or upload file names that are different to local names
    - more complicated code / tests to account for all behaviour paths e.g. optional params specified or not specified, chunking mode and non-chunking mode, etc.
- using the OpenRosa endpoint for submissions
  - pros: most similar to Collect/Enketo(?) client usage
  - cons:
    - would result in a submission that entirely works or doesn't; at least the current code will proceed until the first upload fails or everything works.
    - all other pyodk code uses the Central endpoints rather than OpenRosa
    - not sure how the content-type / chunking works for OpenRosa API usage (presumably at least content-type works)
- including a warning if the provided attachments are more or less than what Central expects
  - pros: could be helpful if a user forgot to include files (no error from Central) or tried to send unreferenced files (this would error 404 from Central)
  - cons: not sure if it's a realistic or common occurrence
- including a warning if multiple files have the same name (e.g. in different directory paths)
  - pros: could be helpful if a user doesn't realise that this happened and/or that it results in overwrites and the most recently submitted file will be saved
  - cons: not sure if it's a realistic or common occurrence
- not adding the blocksize code to session.py
  - pros: less code
  - cons:
    - less obvious what the blocksize is (took a bit of digging to find, ultimately it's defined by urllib3)
    - makes it easier to get the max transmit blocksize consistent with the local file buffer chunk size

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

New functionality requested in #102

#### Do we need any specific form for testing your changes? If so, please attach one.

Included in new tests.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.

Added detailed docstring to SubmissionService. In particular the point about having to include a file reference in the submission XML, since that confused me initially and I didn't see that documented elsewhere.

#### Before submitting this PR, please make sure you have:
- [x] included test cases for core behavior and edge cases in `tests`
- [x] run `python -m unittest` and verified all tests pass
- [x] run `ruff format pyodk tests` and `ruff check pyodk tests` to lint code
- [x] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

**Comment by lindsay-stevens** [comment]
Force pushed for rebase on master (dd5787a4) and applied #111 to `submission_attachments.py`.

File: pyodk/_endpoints/submissions.py [hunk 1]
```diff
@@ -213,14 +220,26 @@ def create(
...
 
+        If provided, each attachment file name must match a file name reference (just the
+        file name, not a directory path) in the submission XML, otherwise Central will
+        reject the upload since the file is not expected. For example, an `image` question
+        named `bird` with a XML value `<bird>pigeon.jpg</bird>` will allow uploading an
+        attachment named `pigeon.jpg`. Central assumes that clients (you), will ensure
+        unique file name references in the submission XML if necessary, e.g. Central will
+        only store one `pigeon.jpg` file across all questions in a given submission. You
+        may provide directory paths to pyodk e.g. `["/home/u/pigeon.jpg"]`, but only the
+        file name is used to identify the file in the upload to Central.
```
Comment (lognaturel) [inline pyodk/_endpoints/submissions.py:235]: This is helpful, I agree. What do you think about also mentioning that chunk size can be set on the session? Users might not realize that's available/helpful in this context without a nudge.

**Review by lognaturel — APPROVED** [review]
One small docs suggestion but looks great otherwise!

**Comment by lognaturel** [comment]
I'll follow up with that tiny docs addition!
